plugins/outputs/influxdb_v2/influxdb.go

```diff
@@ -164,37 +176,53 @@ func (i *InfluxDB) Write(metrics []telegraf.Metric) error {
 			return nil
 		}
 
+		if apiError, ok := err.(*BucketNotFoundError); ok {
+			if i.CreateBuckets {
+				if err := client.CreateBucket(ctx, apiError.Bucket); err != nil {
+					i.Log.Errorf("When writing to [%s]: bucket %q not found and failed to recreate", client.URL(), apiError.Bucket)
+				}
+			}
+		}
+
 		i.Log.Errorf("When writing to [%s]: %v", client.URL(), err)
 	}
 
 	return err
 }
 
-func (i *InfluxDB) getHTTPClient(url *url.URL, proxy *url.URL) (Client, error) {
+func (i *InfluxDB) getHTTPClient(ctx context.Context, u *url.URL, proxy *url.URL) (Client, error) {
 	tlsConfig, err := i.ClientConfig.TLSConfig()
 	if err != nil {
 		return nil, err
 	}
 
 	config := &HTTPConfig{
-		URL:              url,
-		Token:            i.Token,
-		Organization:     i.Organization,
-		Bucket:           i.Bucket,
-		BucketTag:        i.BucketTag,
-		ExcludeBucketTag: i.ExcludeBucketTag,
-		Timeout:          time.Duration(i.Timeout),
-		Headers:          i.HTTPHeaders,
-		Proxy:            proxy,
-		UserAgent:        i.UserAgent,
-		ContentEncoding:  i.ContentEncoding,
-		TLSConfig:        tlsConfig,
-		Serializer:       i.newSerializer(),
+		URL:                    u,
+		Token:                  i.Token,
+		Organization:           i.Organization,
+		Bucket:                 i.Bucket,
+		BucketTag:              i.BucketTag,
+		ExcludeBucketTag:       i.ExcludeBucketTag,
+		CreateBuckets:          i.CreateBuckets,
+		DefaultBucketRetention: int64(time.Duration(i.DefaultBucketRetention).Seconds()),
+		Timeout:                time.Duration(i.Timeout),
+		Headers:                i.HTTPHeaders,
+		Proxy:                  proxy,
+		UserAgent:              i.UserAgent,
+		ContentEncoding:        i.ContentEncoding,
+		TLSConfig:              tlsConfig,
+		Serializer:             i.newSerializer(),
 	}
 
 	c, err := NewHTTPClient(config)
 	if err != nil {
-		return nil, fmt.Errorf("error creating HTTP client [%s]: %v", url, err)
+		return nil, fmt.Errorf("error creating HTTP client [%s]: %v", u, err)
+	}
+
+	if i.CreateBuckets {
+		if err := c.CreateBucket(ctx, c.Bucket); err != nil {
```

Comment:
Do we need to call createbucket here? If we handle the error in Write, this seems unneeded.